src/app.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+//Someandsome
+function createApp() {
+  return new (class {
+    constructor() {
+      this.chitterApi = new chitterApi();
+      this.createViews = new CreateViews();
+      this.allPeeps = this.callPeeps();
+      this.owner = "";
+      this.userId = "";
+      this.sessionKey = "";
+    }
+
+    async callPeeps() {
+      this.createViews.lastPeepCounter = 0;
+      this.createViews.clearPeepsContainer();
+      return await this.chitterApi.fetchAll().then((peeps) => {
+        return peeps.map((peep) => {
+          this.createViews.add(peep, this.owner);
+          return peep;
+        });
+      });
+    }
+
+    async createUser(_user, _password) {
+      this.chitterApi.createUser(_user, _password).then((response) => {
+        if (Array.isArray(response.handle)) {
+          this.createViews.reportFailure(
+            "#registerModalLabel",
+            response.handle[0]
+          );
+        } else {
+          this.createViews.reportSuccessRegistration(
+            "#registerModalLabel",
+            "#regisModalForm",
+            response.handle
+          );
+        }
+      });
+    }
+
+    async loginUser(_user, _password) {
+      this.owner = _user;
+      this.chitterApi.loginUser(_user, _password).then((response) => {
+        if ("errors" in response) {
+          this.owner = "";
+          this.userId = "";
+          this.sessionKey = "";
+          this.createViews.reportFailure(
+            "#loginModalLabel",
+            "and password do not match"
+          );
+        } else {
+          this.userId = response.user_id;
+          this.sessionKey = response.session_key;
+          document.body.setAttribute("data-loggedin", "true");
+          this.createViews.hideModal("#loginModalForm");
+          this.callPeeps();
+        }
+      });
+    }
+
+    async createPeep(_peep) {
```

Comment:
I thought initially I would add a "peep created" but then It came up that adding the peep would trigger the reload of all perps instead of just adding the new one to the top to maintain the 50 peeps limit